analyzers/src/SonarAnalyzer.Common/Helpers/FilesToAnalyzeStorage.cs

```diff
@@ -0,0 +1,48 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2021 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace SonarAnalyzer.Helpers
+{
+    public class FilesToAnalyzeStorage
+    {
+        private readonly Lazy<IEnumerable<string>> filestoAnalyze;
+
+        [ExcludeFromCodeCoverage]
```

Comment:
this only works at class level

Reply:
I wanted to separate concerns and  use  the other constructor and to have unit tests with only this unit but the  other class is removed as is the other constructor so this one is now tested.

Reply:
Should we create an issue to configure it in our build then?

Reply:
I think the answer is yes, as there are a few places where it is used on method level.